packages/storage-plugin/tests/storage.plugin.fallback.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { TestBed } from '@angular/core/testing';
+import { Injectable } from '@angular/core';
+
+import { NgxsModule, State, Store, Action, StateContext } from '@ngxs/store';
+import { NgxsStoragePluginModule } from '..';
+import { DEFAULT_STATE_KEY, fallbackStorage } from '../src/internals';
+
+describe('NgxsStoragePlugin-StorageFallback', () => {
+  class Increment {
+    static type = 'INCREMENT';
+  }
+
+  interface CounterStateModel {
+    count: number;
+  }
+
+  @State<CounterStateModel>({
+    name: 'counter',
+    defaults: { count: 0 }
+  })
+  @Injectable()
+  class CounterState {
+    @Action(Increment)
+    increment({ getState, setState }: StateContext<CounterStateModel>) {
+      setState({
+        count: getState().count + 1
+      });
+    }
+  }
+
+  beforeEach(() => {
+    disableLocalStorage();
+    disableSessionStorage();
+  });
+
+  it('storage engine should work with fallback implementation, when localStorage is not available', () => {
+    // Arrange
+    disableLocalStorage();
+    const spyWarn = jest.spyOn(console, 'warn');
+    TestBed.configureTestingModule({
+      imports: [
+        NgxsModule.forRoot([CounterState]),
+        NgxsStoragePluginModule.forRoot({
+          key: CounterState
+        })
+      ]
+    });
+
+    // Act
+    const store: Store = TestBed.inject(Store);
+    store.dispatch(new Increment());
+    store.dispatch(new Increment());
+    const state: CounterStateModel = store.selectSnapshot(CounterState);
+
+    // Assert
+    expect(spyWarn).toHaveBeenCalledTimes(1);
+    expect(state.count).toBe(2);
+  });
+
+  it('storage engine should work with fallback implementation, when sessionStorage is not available', () => {
+    // Arrange
+    disableSessionStorage();
```

Comment:
Not sure if jest automatically restores the `window` object after each test, but I think it would be safer if we wrapped this test with a `try...finally` and restore the original property on `window` afterwards.

Reply:
it does automatically restore based on my tests. I also tried to use a finally block